src/main/java/com/google/sps/data/ManhattanDistancePathFinder.java

```diff
@@ -0,0 +1,26 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+package com.google.sps.data;
+
+public class ManhattanDistancePathFinder implements PathFinder {
+  public int manhattanDistance(Point A, Point B) {
+    return (int)(Math.abs(A.latitude - B.latitude) + Math.abs(A.longitude - B.longitude));
```

Comment:
`ManhattanDistancePathFinder` implements `PathFinder` interface and it's only used for testing. I use int distance because `GoogleMapsPathFinder.getTimeBetweenPoints(a, b) `, the PathFinder used in production computes the `int` number of seconds to get from point a to point b.

Is there a better way to mock `GoogleMapsPathFinder.getTimeBetweenPoints(a, b) `?

Reply:
I see.

If this function is just used to calculate the "cost" of one route vs another, it might be worth keeping the precision of floating point lat longs and just multiplying them by say 100000 to get a larger int that keeps the precision.

So e.g. the lat lng of two of Google's London offices 6PS and CSG:
* 6PS 51.5206431,-0.135168
* CSG 51.5134121,-0.1407977

So we end up with (51.5206431 - 51.5134121) + (-0.135168 - -0.1407977) = 0.012860700000004416 which as an int is 0 manhattan distance but they are ~2km apart straight-line distance in the real world

If we multiplied those lat longs up before doing the we'd keep some of the precision:
* 6PS 5152064.31,−13516.8,
* CSG 5151341.21,-140797.7

This gives us a more "realistic" manhattan distance/cost of 1286, and your routing will be able to better distinguish things close together in an urban environment.

Up to you - I guess it depends how close all of the points will be in your application.

Reply:
You're right, it makes a lot of sense. I changed the distance function to return the value multiplied by 100000 and updated the tests.